### ISSUE TITLE
add comparison of data via `compare` config

### ISSUE THREAD

**Issue by darrenjennings** [Kong/swrv#41]
`swr` uses a compare function to detect changes to see if the cache needs to be updated. Would be a nice addition before setting the cache. https://github.com/zeit/swr/pull/219

**Comment by bikingbadger** [comment]
Trying to get involved in open source and would like to tackle this one. Might just need to pick your brain a bit to exactly understand what you mean. From what I understand you want the same functionality of using compare in `swrv` as it is in `swr`?

**Comment by darrenjennings** [comment]
@bikingbadger yes we are looking to have some parity re: https://github.com/vercel/swr/pull/655

1. `yarn add dequal`
2. add compare as a function to the config https://github.com/Kong/swrv/blob/master/src/use-swrv.ts#L43 with the default of `import { dequal } from 'dequal/lite`
3. compare the values in the cache before setting the data here: https://github.com/Kong/swrv/blob/master/src/use-swrv.ts#L234
4. add a test `should allow use custom isEqual method` similar to: https://github.com/vercel/swr/blame/a9c10f74963bbe3b566201ef9f43a2242440f0f1/test/use-swr-refresh.test.tsx#L250

**Comment by bikingbadger** [comment]
Great. Thanks for the run down. I will start taking a look at it and working on a solution. Appreciate the help @darrenjennings

**Comment by Danny-Devs** [comment]
Hi @darrenjennings  and @bikingbadger , I've implemented this feature in [PR #375](https://github.com/Kong/swrv/pull/375). The PR adds a custom compare function option and improves the default comparison using dequal/lite. This matches the functionality in the original SWR library while maintaining backward compatibility.

I've added thorough unit tests and plan to verify with the example apps as well.

Looking forward to your review and feedback. Happy to make any adjustments needed!
